mca/identifiers.py

```diff
@@ -83,7 +103,31 @@ def get_discovery_flow_ids(self, hop, ip):
 
         return unique_flow_ids
 
-    def create_new_discovery_flow_id(self, hop, ignore=[]):
+    def create_new_discovery_flow_id(self,
+                                     hop: int,
+                                     ignore: Sequence[Sequence[int]] = None) -> tuple[int]:
+        """
+        Creates a single new flow id for a given hop, based
+        on previous knowledge of existing flow ids for the
+        hop (flow_ids_by_hop) and a list of flow ids to ignore.
+
+        Args:
+            hop (int): hop for which a new flow id will be created.
+            ignore (Sequence[Sequence[int]]): sequence of flow ids to ignore when
+                creating a new flow id.
+
+        Returns:
+            tuple[int]: the new flow id, one int per field.
+
+        Examples:
+            This example assumes that 3 fields are being used:
+            >>> create_new_discovery_flow_id(0, [(1,1,1), (2,2,2), (3,3,3)])
+            (4,4,4)
+        """
+
+        if ignore is None:
```

Comment:
Should we just set the default value for `ignore = ()` on the function declaration to avoid this?